centreon/www/class/centreon-clapi/centreonAPI.class.php

```diff
@@ -716,7 +716,7 @@ public function initXML()
      */
     public function launchAction($exit = true)
     {
-        $action = strtoupper($this->action);
+        $action = htmlspecialchars(strtoupper($this->action), ENT_QUOTES, 'UTF-8');
```

Comment:
I understand the legacy point of view, but I find weird to call htmlspecialchars here because I cannot find any code which set this property outside of this class.

Reply:
@abuathier All these were detected by snyk, the might be dead we just need to check.